ddtrace/tracer/spancontext.go

```diff
@@ -326,11 +326,30 @@ func (t *trace) finishedOne(s *span) {
 	atomic.AddInt64(&tr.spansFinished, int64(len(t.spans)))
 	sd := samplingDecision(atomic.LoadInt64((*int64)(&t.samplingDecision)))
 	if sd != decisionKeep {
+		// if trace sampling decision is drop, we still want to send single spans
+		// unless there are no single span sampling rules defined
+		var singleSpans []*span
```

Comment:
Why not just `spans`?
```suggestion
		var spans []*span
```